Lib/multiprocessing/connection.py

```diff
@@ -286,7 +301,13 @@ def _send_bytes(self, buf):
                 ov.cancel()
                 raise
             finally:
+                self._send_ov = None
                 nwritten, err = ov.GetOverlappedResult(True)
+            if err == WSA_OPERATION_ABORTED:
+                # close() was called by another thread while
+                # WaitForMultipleObjects() was waiting for the overlapped
+                # operation.
+                raise OSError(errno.EPIPE, "handle is closed")
```

Comment:
I chose to raise a ``BrokenPipeError`` exception here, since ``Queue._feed()`` has a special code path for that to ignore EPIPE errors silently:

https://github.com/python/cpython/blob/1ec45378e97a2d3812ed8bc18e1bd1eb8ddcc9a0/Lib/multiprocessing/queues.py#L255-L257

And concurrent.futures uses this code path for its "call queue" which is causing troubles here:

https://github.com/python/cpython/blob/1ec45378e97a2d3812ed8bc18e1bd1eb8ddcc9a0/Lib/concurrent/futures/process.py#L724-L732

Reply:
sounds like we got lucky that callers were handling one thing we could raise! :)

Reply:
At the beginning, I started by adding a new exception. But I chose to reuse the existing code instead. IMO BrokenPipeError perfectly makes sense for a **Pipe**Connection.